app/Models/Event.php

```diff
@@ -528,8 +532,10 @@ private static function beatmapsetParams($beatmapset)
     private static function beatmapParams($beatmap, $ruleset)
     {
         $url = e(route('beatmaps.show', ['beatmap' => $beatmap, 'ruleset' => $ruleset], false));
+        $title = "{$beatmap->beatmapset->artist} - {$beatmap->beatmapset->title} [{$beatmap->version}]";
         return [
-            'title' => e("{$beatmap->beatmapset->artist} - {$beatmap->beatmapset->title} [{$beatmap->version}]"),
+            'title' => e($title),
+            'title_clean' => $title,
```

Comment:
or actually, considering how they end up being used just to generate links and that none of these being reused due to differing escaping, just return `'link' => tag('a', ...), 'link_clean' => "[{$url} {$title}]"` for all those params thing instead...? and call the functions `thingLink` or something

each of these functions can just first generate the title and url and pass it to a function which generate both link variants

one weird exception looks like the username change which uses a different name but accepting an additional parameter for the function may be workable (or some other methods)

there are a few places where `$user['id']` is being used but it's easily available through a different mean anyway.